jsonnet-mode.el

```diff
@@ -248,21 +250,32 @@ If not inside of a multiline string, return nil."
 (defun jsonnet-eval-buffer ()
   "Run jsonnet with the path of the current file."
   (interactive)
-  (let ((file-to-eval (buffer-file-name))
-        (search-dirs jsonnet-library-search-directories))
-    (when (buffer-modified-p)
-      (when (y-or-n-p
-             (format "Save file %s? " file-to-eval))
-        (save-buffer)))
-    (with-current-buffer (get-buffer-create "*jsonnet output*")
+  (let ((file-to-eval (file-truename (buffer-file-name)))
+        (search-dirs jsonnet-library-search-directories)
+        (output-buffer-name "*jsonnet output*"))
+    (save-some-buffers (not compilation-ask-about-save)
+                       (lexical-let ((directories (cons (file-name-directory file-to-eval)
+                                                        search-dirs)))
+                         (lambda ()
+                           (member (file-name-directory (file-truename (buffer-file-name)))
+                                   directories))))
+    (when-let ((output-window (get-buffer-window output-buffer-name t)))
+      (quit-window nil output-window)
+      (redisplay))
+    (with-current-buffer (get-buffer-create output-buffer-name)
+      (setq buffer-read-only nil)
       (erase-buffer)
       (let ((args (nconc (cl-loop for dir in search-dirs
                                   collect "-J"
                                   collect dir)
                          (list file-to-eval))))
-        (apply #'call-process jsonnet-command nil t nil args))
-      (when (fboundp 'json-mode)
-        (json-mode))
+        (if (zerop (apply #'call-process jsonnet-command nil t nil args))
```

Comment:
See google/jsonnet#638 for a proposed change to how _jsonnet eval_ writes its error output.